# Problem 10.1
In this problem I was to chop function.

In [19]:
def chop2(list):
    del list[0]
    del list[0]
    print(list)
    

In [20]:
chop2([1,2,3,4])

[3, 4]


**Solution**: To solve this problem I created a function that takes a list as a parameter and chops the first two values.

# Problem 10.2
In this problem I was to take a string and convert it.

In [37]:
spam = ['apples', 'bananas', 'tofu', 'cats']

def comma_list(list):
    return ', '.join(list[:-1] + ['and '+list[-1]])

comma_list(spam)


'apples, bananas, tofu, and cats'

**Solution**: To solve this program I used the join() function to have it return as one string.

# Problem 10.3
For this problem I was to a function that takes a list and does a specific task.

In [24]:
def d_nested_sum(numbers):
    total = 0
    for number in numbers:
        for num in number:
            total += num
    return total / 2
    

In [38]:
t = [[1, 2], [3], [4, 5, 6]]
d_nested_sum(t)

10.5

**Solution:** To solve this problem I created a for loop within a for loop.

# Problem 10.4
In this problem I was to creat a character grid with loops.

In [26]:
grid=[['.', '.', '.', '.', '.', '.'],
      ['.', 'O', 'O', '.', '.', '.'],
      ['O', 'O', 'O', 'O', '.', '.'],
      ['O', 'O', 'O', 'O', 'O', '.'],
      ['.', 'O', 'O', 'O', 'O', 'O'],
      ['O', 'O', 'O', 'O', 'O', '.'],
      ['O', 'O', 'O', 'O', '.', '.'],
      ['.', 'O', 'O', '.', '.', '.'],
      ['.', '.', '.', '.', '.', '.']]


for i in range(len(grid[0])):
    for j in range(len(grid)):
        print(grid[j][i], end=' ')
    print('')

. . O O . O O . . 
. O O O O O O O . 
. O O O O O O O . 
. . O O O O O . . 
. . . O O O . . . 
. . . . O . . . . 


**Solution:** To solve this problem, I used a loop within a loop. 

# Problem 10.5
In this problem I was to create a function that takes tuples and returns their product.

In [34]:
def prod_all(*numbers):
    tupl_sum = (numbers)
    total = 1
    for i in tupl_sum:
        total *= i
    return total

print(prod_all(1, 2, 3, 4))
print(prod_all(4, 6, 8))

24
192


**Solution:** To solve this problem, I created the function prod_all that takes a tuple argument and multiples them by each other.

# Problem 10.6
To solve this problem I was to creat a function called `cumsum_twice()` that takes a list of numbers and returns twice the cumulative sum

In [36]:
def cumsum_twice(numbs):
    c_sum = []
    total = 0
    for i in numbs:
        total += i
        c_sum.append(2*total)
    return c_sum

cumsum_twice([1, 2, 3])

[2, 6, 12]

# Problem 10.7 (Video at Bottom)
In this problem I was to use the CPX board to display various intensities of light

```Python
import time
from adafruit_circuitplayground import cp

intensities = [0.004, 0.006, 0.01, 0.02, 0.04, 0.08, 0.16, 0.32, 0.64, 1.0]

while True:
    length = len(intensities)
    for i in range(length):
        cp.pixels[i] = (255*intensities[i],0,0)
        time.sleep(0.5)
        if i == length:
            continue
    for i in range(length):
        cp.pixels[-i] = (0,255*intensities[-i],0)
        time.sleep(0.5)
    for i in range(length):
        cp.pixels[i] = (0,0,255*intensities[i])
        time.sleep(0.5)
```

**Solution:** To solve this problem I used for loops.

# Problem 10.8 (Video at Bottom)
In this problem I was to have the CPX board play a sound based on the acceleration.

```Python 
import time
from adafruit_circuitplayground import cp
import math

while True:
    time.sleep(0.2)
    x, y, z = cp.acceleration
    mag_acc = math.sqrt(x*x + y*y + z*z)
    print("Acceleration:", mag_acc)
    cp.play_tone(mag_acc, 0.5)


**Solution:** I had it play a tone based on the value of acceleration.

# Problem 10.9 (Video at Bottom)
In this problem I was to have the lights change based on acceleration.

```Python
import time
from adafruit_circuitplayground import cp

while True:
    time.sleep(0.2)
    x, y, z = cp.acceleration
    newX = int(abs(x))
    newY = int(abs(y))
    newZ = int(abs(z))
    print(newX, newY, newZ)
    cp.pixels.fill((newX, newY, newZ))
```
**The reason the lights are still on while the board is flat on the table is because there is still the acceleration of earths gravity which is around 9.8**

**Solution** To solve this problem I first converted my values into positive integers then I inputed them as pixel values.

# Video 10.7-10.9

https://youtu.be/bWoPjLi9Q7E

https://img.youtube.com/vi/bWoPjLi9Q7E/0.jpg

<img src="hw_10_pic.jpg" width="400" height="400">